### DISCUSSION TITLE
python3-APScheduler_3.10.4.bb

### DISCUSSION THREAD

**Discussion by p35420102** [agronholm/apscheduler#852] [Q&A]
Is python3-APScheduler_3.10.4.bb provided?
I currently have usage requirements on yocto

**Comment by agronholm** [comment]
What are you talking about?

**Reply by p35420102** [reply]
Oh...just literally?
I've searched all over and can't find any .bb files related to APScheduler.

**Reply by agronholm** [reply]
Am I supposed to know what a .bb file is?

**Reply by p35420102** [reply]
Is used by yocto
For example this python3-psutil_5.9.6.bb
![IMG_6496](https://github.com/agronholm/apscheduler/assets/47807222/906ade24-8fa6-4b9f-b94a-9143a923162d)


But I searched all over the Internet and there is no relevant webpage of APScheduler with .bb.

**Reply by agronholm** [reply]
Then I guess it doesn't exist? This is the repository of the upstream project. Distro packaging happens elsewhere.

**Reply by p35420102** [reply]
Is there a way to download it through pypi now?

**Reply by agronholm** [reply]
Download what? APScheduler? That's always been possible.

**Reply by p35420102** [reply]
I tried to write out the recipe but it seems that it cannot be compiled successfully on yocto? 
I also use pypi setuptools3.

![apscheduler](https://github.com/agronholm/apscheduler/assets/47807222/878a0214-fda1-475e-ba7b-f774937b47d8)

Build Configuration:
BB_VERSION           = "2.0.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-poky-linux"
MACHINE              = "genio-350-evk"
DISTRO               = "rity-demo"
DISTRO_VERSION       = "23.1-release"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-poky            
meta-yocto-bsp       = "HEAD:1e6f147c686a91b9a5996f529e32ee1852f946b7"
meta-arm             
meta-arm-toolchain   = "HEAD:96aad3b29aa7a5ee4df5cf617a6336e5218fa9bd"
meta-clang           = "HEAD:71321ddf78ea522b87a6b4bffefb14c988a6d921"
meta-mediatek-bsp    = "HEAD:cbdeb08971c7e705f909351340e29977cf30356e"
meta-mediatek-tsn    = "HEAD:ad9bddbf1e9d4e56e94e93de61fee9d347e451e8"
meta-mediatek-bluedroid = "HEAD:71ae5cfbc089d1a1c64e3e76e316f4982635cb1a"
meta-mediatek-demo   = "HEAD:399ef9563e2ba6d887771138c01ab5aaad164b18"
meta-nn              = "HEAD:bed2b21cef6ef874ef88d8bcc5429970a2b9ee79"
meta-stress-test     = "HEAD:534b6122c2462d4da14aad42ee42b50d3301ecd9"
meta-security        = "HEAD:cc20e2af2ae1c74e306f6c039c4963457c4cbd0f"
meta-multimedia      
meta-networking      
meta-oe              
meta-python          
meta-perl            = "HEAD:5f120a926b0fcd55cfe7565bb7ddf23661cad498"
meta                 
meta-rity-bringup    
meta-rity-bsp        
meta-rity-demo       = "HEAD:2b3f3f14daa695469025086fe638da8ebe4a848e"
meta-tensorflow      = "HEAD:ec20e19319f9eb89ceadc04923cc5bc75e865692"
meta-qt5             = "HEAD:bff5bd937f0776166e81a63f3dd39ede348ef758"
workspace            = "<unknown>:<unknown>"

WARNING: /home/gbt/project/iot-yocto/src/poky/../meta-mediatek-bsp/recipes-kernel/linux/linux-mtk_5.15.bb:do_compile is tainted from a forced run
Initialising tasks: 100% |############################################################| Time: 0:00:02
Sstate summary: Wanted 1067 Local 1053 Mirrors 0 Missed 14 Current 2820 (98% match, 99% complete)
Removing 2 stale sstate objects for arch genio_350_evk: 100% |########################| Time: 0:00:00
NOTE: Executing Tasks
ERROR: python3-apscheduler-3.10.4-r0 do_compile: 'python3 setup.py bdist_wheel ' execution failed.
ERROR: python3-apscheduler-3.10.4-r0 do_compile: ExecutionError('/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/temp/run.do_compile.555623', 1, None, None)
ERROR: Logfile of failure stored in: /home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/temp/log.do_compile.555623
Log data follows:
| DEBUG: Executing shell function do_compile
| /home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/dist.py:720: UserWarning: Usage of dash-separated 'upload-dir' will not be supported in future versions. Please use the underscore name 'upload_dir' instead
|   warnings.warn(
| /home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/installer.py:27: SetuptoolsDeprecationWarning: setuptools.installer is deprecated. Requirements should be satisfied by a PEP 517 installer.
|   warnings.warn(
| /home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/bin/python3-native/python3: No module named pip
| Traceback (most recent call last):
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/installer.py", line 82, in fetch_build_egg
|     subprocess.check_call(cmd)
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/subprocess.py", line 369, in check_call
|     raise CalledProcessError(retcode, cmd)
| subprocess.CalledProcessError: Command '['/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/bin/python3-native/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmp1o7f3uli', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.
| 
| The above exception was the direct cause of the following exception:
| 
| Traceback (most recent call last):
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/APScheduler-3.10.4/setup.py", line 10, in <module>
|     setup(
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/__init__.py", line 152, in setup
|     _install_setup_requires(attrs)
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/__init__.py", line 147, in _install_setup_requires
|     dist.fetch_build_eggs(dist.setup_requires)
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/dist.py", line 809, in fetch_build_eggs
|     resolved_dists = pkg_resources.working_set.resolve(
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 771, in resolve
|     dist = best[req.key] = env.best_match(
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1056, in best_match
|     return self.obtain(req, installer)
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/pkg_resources/__init__.py", line 1068, in obtain
|     return installer(requirement)
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/dist.py", line 880, in fetch_build_egg
|     return fetch_build_egg(self, req)
|   File "/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/lib/python3.10/site-packages/setuptools/installer.py", line 84, in fetch_build_egg
|     raise DistutilsError(str(e)) from e
| distutils.errors.DistutilsError: Command '['/home/gbt/project/iot-yocto/build/tmp/work/armv8a-poky-linux/python3-apscheduler/3.10.4-r0/recipe-sysroot-native/usr/bin/python3-native/python3', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmp1o7f3uli', '--quiet', 'setuptools_scm']' returned non-zero exit status 1.
| ERROR: 'python3 setup.py bdist_wheel ' execution failed.
| WARNING: exit code 1 from a shell command.
ERROR: Task (/home/gbt/project/iot-yocto/src/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-apscheduler_3.10.4.bb:do_compile) failed with exit code '1'
NOTE: Tasks Summary: Attempted 7094 tasks of which 7093 didn't need to be rerun and 1 failed.

Summary: 1 task failed:
  /home/gbt/project/iot-yocto/src/poky/../meta-openembedded/meta-python/recipes-devtools/python/python3-apscheduler_3.10.4.bb:do_compile
Summary: There was 1 WARNING message.
Summary: There were 2 ERROR messages, returning a non-zero exit code.

**Reply by agronholm** [reply]
Whatever the case, I can't help you with distro specific packaging. You should go to a distro specific forum for assistance.
